shared/models/notification.rb

```diff
@@ -0,0 +1,69 @@
+require "securerandom"
+
+module FastlaneCI
+  # All metadata about a notification.
+  class Notification
+    # Is a UDID so we're not open to ID guessing attacks
+    #
+    # @return [String]
+    attr_accessor :id
+
+    # The relative priority of a notification
+    #
+    # @return [String]
+    attr_accessor :priority
+
+    # The name of a notification, which combined with the message can uniquely
+    # specify it
+    #
+    # @return [String]
+    attr_accessor :name
+
+    # Notification message to be displayed in the dashboard
+    #
+    # @return [String]
+    attr_accessor :message
+
+    # The time the notification was created
+    #
+    # @return [Time]
+    attr_accessor :created_at
+
+    # The last time the notification was updated
+    #
+    # @return [Time]
+    attr_accessor :updated_at
+
+    # A way to uniquely specify a notification
+    #
+    # @return [String]
+    attr_accessor :primary_key
+
+    # Instantiates a new `Notification` model object
+    #
+    # @param  [String] id
+    # @param  [String] priority
+    # @param  [String] name
```

Comment:
Could a message be user specific also? I could imagine things like:
- Invalid scope for your GitHub user account
- [Maybe] a build you triggered has failed
Maybe there is more in the future?

Reply:
I was thinking of associating notifications to users by persisting the user `email` in `notifications.json`. That way the `email` attribute acts as a foreign key. Does this sound like an okay approach, @taquitos. Felix said you wrote the user-specific logic, so I wanted to run the idea by you.

I saw `email.downcase` was used for comparisons when determining if a user already exists - so it acts as a primary key for the `User` model.

Reply:
Yeah, that's good for determining if a user already exists, compare all the emails they want to use.
But, for comparing anywhere else, we should use the `id` as the foreign key.